crates/data_components/src/poly.rs

```diff
@@ -0,0 +1,111 @@
+use std::{any::Any, sync::Arc};
+
+use arrow::datatypes::SchemaRef;
+use async_trait::async_trait;
+use datafusion::{
+    catalog::Session,
+    common::Constraints,
+    datasource::{TableProvider, TableType},
+    error::{DataFusionError, Result as DataFusionResult},
+    logical_expr::LogicalPlan,
+    physical_plan::ExecutionPlan,
+    prelude::Expr,
+};
+use datafusion_federation::{FederatedTableProviderAdaptor, FederationProvider};
+
+use crate::delete::{get_deletion_provider, DeletionTableProvider};
+
+#[derive(Clone)]
+pub struct PolyTableProvider {
+    write: Arc<dyn TableProvider>,
+    delete: Arc<dyn TableProvider>,
+    fed: Arc<dyn TableProvider>,
```

Comment:
Why not just store the values with their concrete types?

Reply:
This is meant to be handling all types of accelerator. The only thing can be concrete is delete which can use the adaptor. Will change it so.